@xen-orchestra/xapi/vm.js

```diff
@@ -477,21 +510,17 @@ class Vm {
     }
   }
 
-  async snapshot($defer, vmRef, { cancelToken = CancelToken.none, name_label } = {}) {
+  async snapshot($defer, vmRef, { cancelToken = CancelToken.none, ignoreNobakVdis = false, name_label } = {}) {
     const vm = await this.getRecord('VM', vmRef)
+
     // cannot unplug VBDs on Running, Paused and Suspended VMs
-    if (vm.power_state === 'Halted' && this._ignoreNobakVdis) {
-      await asyncMap(vm.VBDs, async vbdRef => {
-        const vbd = await this.getRecord('VBD', vbdRef)
-        if (
-          vbd.type === 'Disk' &&
-          Ref.isNotEmpty(vbd.VDI) &&
-          (await this.getField('VDI', vbd.VDI, 'name_label')).startsWith('[NOBAK]')
-        ) {
-          await this.VBD_destroy(vbdRef)
-          $defer.call(this, 'VBD_create', vbd)
-        }
+    if (ignoreNobakVdis && vm.power_state === 'Halted') {
+      await asyncMap(await listNobakVbds(this, vm.VBDs), async vbd => {
+        await this.VBD_destroy(vbd.$ref)
+        $defer.call(this, 'VBD_create', vbd)
       })
+
+      ignoreNobakVdis = false
```

Comment:
I would prefer a dedicated variable like noBakVdiHandled instead of repurposing ignoreNoBakVdi